### PR TITLE
Add file size check to CI

### PR DESCRIPTION
**Pull request by anthonydillon**
## Done
Add filesize check to CI with a threshold

## QA
- Run `./run clean`
- Run `tests/file-size-lint.sh` and see that it errors because it cant find the build file
- Run `./run build`
- Run `tests/file-size-lint.sh` and see it passes and outputs the current file size of the build with the threshold

_The threshold is set to 400kb which is way to high. Lets half that and more as soon as possible_

### CODE REVIEW

**Comment by anthonydillon** [comment]
@nottrobin so CI failed but not in a way I would like it too :smile:

**Comment by nottrobin** [comment]
@anthonydillon I don't know why you're getting this suddenly, but try adding `gem install rake` to the install steps in `.travis.yaml`

**Comment by WillMoggridge** [comment]
@anthonydillon I'm not entirely sure what started it, current develop branch seems to build just fine.

May be this helps: https://github.com/travis-ci/travis-ci/issues/8978#issuecomment-354035321

**Comment by anthonydillon** [comment]
@nottrobin nope :(

**Comment by anthonydillon** [comment]
@WillMoggridge @nottrobin I'm stumped

**Comment by anthonydillon** [comment]
@nottrobin I have updated the test to be more resilient and should now work with travis.

File: tests/file-size-lint.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+if [ ! -f $FILE ]; then
+  echo 'ERROR: Build file could not be found or is not readable. Please build first with `./run build`'
+  exit 1
+fi
+
+THRESHOLD=400000
+CURRENTSIZE=$(du -b $FILE | cut -f 1)
+MESSAGE="Your generated CSS is ${CURRENTSIZE} with a threshold of ${THRESHOLD}"
+if [ $CURRENTSIZE -lt $THRESHOLD ]; then
+  STATUS='SUCCESS'
```
Comment (nottrobin) [inline tests/file-size-lint.sh]: Why not simply `echo` and exit directly in each of these cases? I know you'll be duplicating the message, but arguably those messages should be slightly different, given the radically different results:

``` bash
if [ ${CURRENTSIZE} -gte ${THRESHOLD} ]; then
  echo "ERROR: Your CSS is over ${THRESHOLD} (it's ${CURRENTSIZE})"
  exit 1
fi

echo "Success: your generated CSS is ${CURRENTSIZE} (under ${THRESHOLD})"
```

File: tests/file-size-lint.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  EXIT=0
+else
+  STATUS='ERROR'
+  EXIT=1
+fi
+
+echo "${STATUS}: ${MESSAGE}"
```
[the remark this answers is not in the document]
Comment (anthonydillon) [reply tests/file-size-lint.sh]: Good idea.

File: tests/file-size-lint.sh (hunk 1 above)
Comment (anthonydillon) [reply tests/file-size-lint.sh]: Good idea

**Comment by anthonydillon** [comment]
@nottrobin done and ready for another look.

**Pull request merged**